lib/reek/smells/smell_repository.rb

```diff
@@ -43,25 +42,21 @@ def examine(scope)
       end
 
       def detectors
-        @initialized_detectors ||= begin
-          @detectors = {}
-          smell_types.each do |klass|
-            @detectors[klass] = klass.new(source_via)
-          end
-          @detectors
-        end
+        @initialized_detectors ||= smell_types.map do |klass|
+          { klass => klass.new(source_via) }
+        end.reduce({}, :merge)
       end
 
-      private
+      protected
```

Comment:
> Strong veto.

Ok, when you two agree on how we should deal with the warnings please let me know and I’ll update this PR accordingly. :)

> We just do this to silence a warning we only see when we run the spec?

No, to silence warnings we otherwise throw at anyone who uses the reek API.